### DISCUSSION TITLE
syncInput

### DISCUSSION THREAD

**Discussion by wr0nggg** [livewire/livewire#6894] [1. Help]
is there a replacement for syncInput from livewire 2?
We used it to parse date inputs to null / carbon but its gone now

**Comment by joshhanley** [comment]
@wr0nggg can you share your code on how you're using it?

**Reply by wr0nggg** [reply]
lw2 -> 
```
class BaseComponent extends Component
{
    public function syncInput($name, $value, $rehash = true)
    {
        if ($value === '') {
            $value = null;
        }
        parent::syncInput($name, $value, $rehash);
    }
}
```
js library for dates after deleting the date send empty string instead of null, this solved it

**Reply by joshhanley** [reply]
@wr0nggg so in V3 we have the concept of synths, which should handle converting most empty strings to null for different property types. If there are any we are missing, PRs are welcome to fix them up! If that doesn't work, I think we should come up with a solution for converting empty strings to null, as that is something most people and inputs will need.

**Reply by ianwinstanley** [reply]
I have a similar problem, having just upgraded from Livewire 2 to Livewire 3. In order to detect a change in a dropdown named "mapper" I override `syncInput`, call the parent, then instantiate a class and load a new schema for the form depending on the `mapper` value. It worked great before and the form just reflowed with the new fields, based on the schema.

Is there any event detection in Livewire 3 that might replace this?

``` 
  public function syncInput($name, $value, $rehash = true)
    {
        parent::syncInput($name, $value);
        if ($name === "mapper") {
            $class = $value;
            $this->schema = $class::getConfigSchema();
        }
    }
```
